src/components/SingleVault/index.js

```diff
@@ -1,86 +1,190 @@
-	const [showHistoricHarvests, setShowHistoricHarvests] = useState([]);
+	const [strategies, setStrategies] = useState([]);
+	const [showHarvestHistory, setShowHarvestHistory] = useState({});
+	const [harvestingAll, setHarvestingAll] = useState(false);
 	const [vault, setVault] = useState({});
-	const [nonce, setNonce] = useState(0);
-	const [harvestedS, setHarvested] = useState([]);
 	const [showRatio, toggleRatios] = useState(false);
 	const [zeros, setStateZeros] = useState({});
 	const [showGraphs, setShowGraphs] = useLocalStorage('SingleVault.settings.showGraphs', false);
-	
-	//console.log('inputting ', value);
-
-	//Handle the setAll
-	// const	onSetAll = useCallback(async () => {
-	// 	console.log('sart');
-	// 	const vlt = await AllVaults(value, provider);
-	// 	console.log(vlt);
-	// 	setVault(vlt);
-	// 	const	_all = await AllStrats(vlt, provider);
-	// 	console.log('changing!');
-	// 	console.log(_all);
-	// 	setAlls(_all || []);
-	// }, [value, provider]);
-	// useEffect(() => onSetAll(), [onSetAll]);
+	const provider = value.chain == 250 ? fantomProvider : defaultProvider;
+	const anyHarvests = strategies.some(s => s.succeded);
 
 	useEffect(() => {
-		
 		if(value.address && provider){
-			AllVaults(value, provider).then(x => {
-				setVault(x);
+			AllVaults(value, provider).then(freshVault => {
+				setVault(freshVault);
 			});
 		}
-		
 	}, [value, provider]);
 
 	useEffect(() => {
-
 		if(vault.address && provider){
-			AllStrats(vault, provider).then(y => {			
-				setAlls(y || []);
-				y.forEach(strategy => {
-					axios.post('api/getVaults/AllStrategyReports', strategy).then((response) => {
-						//console.log(response.data);
-						setHistoricHarvests(
-							currentValues => {
-								currentValues[strategy.address] = response.data;
-								return currentValues;
-							
-							});
-						setNonce(nonce+Math.random(100));
-			
-					});
-				});
-				
+			AllStrats(vault, provider).then(freshStrategies => {
+				freshStrategies.sort((a, b) => a.lastTime - b.lastTime);
+				(async () => {
+					for(const strategy of freshStrategies) {
+						strategy.harvesting = false;
+						const response = await axios.post('api/getVaults/AllStrategyReports', strategy);
+						strategy.harvestHistory = response.data;
+					}
+					setStrategies([...freshStrategies]);
+				})();
 			});
 		}
-
 	}, [vault, provider]);
 
-	function runSimZero(strat){
+	const getStrategyApr = useCallback((strategy) => {
+		if(!strategy.succeded){
+			return {beforeFee: 0, afterFee: 0};
+		}
+
+		const profit = strategy.paramsAfterHarvest.totalGain - strategy.beforeGain;
+		const loss = strategy.paramsAfterHarvest.totalLoss - strategy.beforeLoss;
+
+		let percent = 0;
+		if (strategy.beforeDebt > 0) {
+			if (loss > profit){
+				percent = -1 * loss / strategy.beforeDebt; 
+			} else {
+				percent = profit / strategy.beforeDebt;
+			}
+		}
+
+		const performanceFee = vault.performanceFee / 10_000;
+		const managementFee = vault.managementFee / 100;
```

Comment:
When I was debugging values here, I remember thinking it looked like management fee was in percentage format and performance fee was in bps format. And I thought it might not be right! But I forgot to research or ask about that specifically. Is management fee definitely in bps?

Reply:
Yes, definitely. Current mgmt fee should be 200 which is 0.2 which is 2%.